cephfs/file_xattr.go

```diff
@@ -97,3 +98,43 @@ func (f *File) GetXattr(name string) ([]byte, error) {
 	}
 	return buf[:ret], nil
 }
+
+// ListXattr returns a slice containing strings for the name of each xattr set
+// on the fie.
+//
+// Implements:
+//  int ceph_flistxattr(struct ceph_mount_info *cmount, int fd, char *list, size_t size);
+func (f *File) ListXattr() ([]string, error) {
+	if err := f.validate(); err != nil {
+		return nil, err
+	}
+
+	var (
+		ret C.int
+		err error
+		buf []byte
+	)
+	// range from 1k to 64KiB
+	retry.WithSizes(1024, 1<<16, func(size int) retry.Hint {
+		buf = make([]byte, size)
+		ret = C.ceph_flistxattr(
+			f.mount.mount,
+			f.fd,
+			(*C.char)(unsafe.Pointer(&buf[0])),
+			C.size_t(size))
+		err = getErrorIfNegative(ret)
+		return retry.DoubleSize.If(err == errRange)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	names := make([]string, 0)
```

Comment:
Shouldn't we wait for #313 to get merged?

Reply:
This prompted me to work on that one but I don't think it matters which one lands first. If it this one, I will update #313 to include these functions. If #313 lands first, I will update this PR to make use of it.

Reply:
#313 is merged now.

Reply:
I've pushed a follow up patch to the PR, PTAL.